packages/synapse-react-client/src/components/SynapseChat/GridAgentChat.tsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from 'react'
+import DraggableDialog from '../DraggableDialog/DraggableDialog'
+import { SynapseChat } from './index'
+import { ChatInteraction } from './SynapseChat'
+import { AgentSession } from '@sage-bionetworks/synapse-types'
+
+export type GridAgentChatProps = {
+  gridSessionId: string
+  usersReplicaId: number
+  chatbotName?: string
+  initialMessage?: string
+  open: boolean
+  onClose: () => void
+}
+
+export function GridAgentChat({
+  gridSessionId,
+  usersReplicaId,
+  chatbotName = 'Grid Assistant',
+  initialMessage,
+  open,
+  onClose,
+}: GridAgentChatProps) {
+  const [agentSession, setAgentSession] = useState<AgentSession | undefined>()
+  const [interactions, setInteractions] = useState<ChatInteraction[]>([])
```

Comment:
Just add a comment to clarify why the state was lifted

```suggestion
  // Storing state for the chat session here preserves chat history while the dialog is opened and closed.
  const [agentSession, setAgentSession] = useState<AgentSession | undefined>()
  const [interactions, setInteractions] = useState<ChatInteraction[]>([])
```